src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,400 @@
+- `return jsonify({"task": task})` returns a JSON response containing the updated task.
+
+### DELETE Request
+A DELETE request is used to remove a resource from a server. To delete a task, implement a route that handles DELETE requests:
+```python
+@app.route("/tasks/<int:task_id>", methods=["DELETE"])
+def delete_tasks(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    tasks.remove(task)
+    return jsonify({"result": "Task deleted"}), 200
+```
+- `@app.route` binds a URL pattern `/tasks/<int:task_id>` to the function that follows it.
+- `<int:task_id>` is a variable part of the URL that will be captured and passed to the function as an argument.
+- `methods=["DELETE"]` specifies that this route should respond to the HTTP DELETE requests. This means that when a client sends a DELETE request to the `/tasks/<int:task_id>` URL, Flask will call the associated function.
+- `delete_task()` function will take `task_id` as argument that is extracted from `/tasks/<int:task_id>` URL. The function will then search for a task with that ID in the `tasks` list. If a task is found, it will remove the task from the list and return a JSON response indicating that the task was deleted. If no task is found, it will return a `404` status code with an error message.
+- `tasks.remove(task)` removes the task from the `tasks` list.
+- `return jsonify({"result": "Task deleted"})` returns a JSON response indicating that the task was successfully deleted.
+
+### Running the Application
+To run your application, add the following line:
+```python
+if __name__ == "__main__":
+    app.run(debug=True)
+```
+
+Then open your terminal and run the following command form your project directory:
+```bash
+python3 app.py
+```
+Your REST API is now up and running on `http://127.0.0.1:5000`
+
+## Testing Your API
+You can test your API using tools like Postman or cURL.
+
+### List All Tasks
+You can use the following cURL command to list all the avaliable tasks:
+```bash
+curl http://127.0.0.1:5000/tasks
+```
+Expected Result:
+```json
+{
+  "tasks": [
+    {
+      "completed": false,
+      "id": 1,
+      "title": "Buy groceries"
+    },
+    {
+      "completed": true,
+      "id": 2,
+      "title": "Clean the house"
+    }
+  ]
+}
+```
+
+### List a Specific Task
+To retrieve a specific task by its ID, you can use the following cURL command, replacing `<task_id>` with the id of the task you want to retrieve:
+```bash
+curl http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl http://127.0.0.1:5000/tasks/1
+```
+The `curl` command sends an HTTP GET request to `http://127.0.0.1:5000/tasks/1`, retrieving the task with ID `1` from a local server.
```

Comment:
```suggestion
The cURL command sends an HTTP GET request to `http://127.0.0.1:5000/tasks/1`, retrieving the task with ID `1` from a local server.
```

Reply:
IMO the current one is fine. Just highlighting the command used in the script. If it was in documentation section then `cURL` would be nice.